synapse/cortex.py

```diff
@@ -1619,15 +1616,21 @@ async def getCoreQueue(self, iden):
             return self.multiqueue.status(iden)
         return
 
-    async def reqCoreQueue(self, iden=None, name=None):
-        if name:
-            if (iden := self.quedefs.get(name)) is None:
-                raise s_exc.NoSuchName(mesg=f'No queue with name {name}', name=name)
-
+    async def reqCoreQueue(self, iden):
         if (info := await self.getCoreQueue(iden)) is None:
             raise s_exc.NoSuchIden(mesg=f'No queue with iden {iden}', iden=iden)
         return info
 
+    async def getCoreQueueByName(self, name):
+        if (info := await self.reqCoreQueueByName(name)):
+            return info
+        return
+
+    async def reqCoreQueueByName(self, name):
+        if (iden := self.quedefs.get(name)) is None:
+            raise s_exc.NoSuchName(mesg=f'No queue with name {name}', name=name)
+        return await self.getCoreQueue(iden)
+
```

Comment:
`getCoreQueueByName()` can't call `reqCoreQueueByName()` because it will raise an exception. You had all the pieces but needed a slight tweak :) 

```suggestion
    async def reqCoreQueueByName(self, name):
        if (info := await self.getCoreQueueByName(name)):
            return info
        raise s_exc.NoSuchName(mesg=f'No queue with name {name}', name=name)

    async def getCoreQueueByName(self, name):
        if (iden := self.quedefs.get(name)) is None:
            return None
        return await self.getCoreQueue(iden)

```

Reply:
Thanks, its been a long week :laughing: